Example Code/Reflowduino_Demo/Reflowduino_Demo.ino

```diff
@@ -127,57 +140,95 @@ int noteDurations[] = {4, 8, 8, 4, 4, 4, 4, 4};
 
 // Logic flags
 bool justStarted = true;
-bool reflow = false; // Baking process is underway!
+volatile bool reflow = false; // Baking process is underway!
+volatile bool buttonPressed = false;
+volatile unsigned long lastButtonPress = 0;
+#define BUTTON_DEBOUNCE_MSEC 100
+
 bool preheatComplete = false;
+bool stageWait = false;
 bool soakComplete = false;
-bool reflowComplete = false;
+bool reflowAComplete = false;
+bool reflowBComplete = false;
 bool coolComplete = false;
 
 double T_start; // Starting temperature before reflow process
 int windowSize = 2000;
 unsigned long sendRate = 2000; // Send data to app every 2s
 unsigned long t_start = 0; // For keeping time during reflow process
 unsigned long previousMillis = 0;
+unsigned long previousDebugMillis = 0;
+
 unsigned long duration, t_final, windowStartTime, timer;
 
 void setup() {
   Serial.begin(9600); // This should be different from the Bluetooth baud rate
   BT.begin(57600);
+  delay(10); // wait a bit before initting
 
   pinMode(buzzer, OUTPUT);
   pinMode(LED, OUTPUT);
   pinMode(relay, OUTPUT);
 
+  digitalWrite(BTN_P2, INPUT_PULLUP);
+  digitalWrite(BTN_P3, INPUT_PULLUP);
+  attachInterrupt(digitalPinToInterrupt(BTN_P2), buttonPress, FALLING);
+  attachInterrupt(digitalPinToInterrupt(BTN_P3), buttonPress, FALLING);
+
   digitalWrite(LED, LOW);
   digitalWrite(relay, LOW); // Set default relay state to OFF
 
   myPID.SetOutputLimits(0, windowSize);
   myPID.SetSampleTime(PID_sampleTime);
   myPID.SetMode(AUTOMATIC); // Turn on PID control
 
-//   while (!Serial) delay(1); // OPTIONAL: Wait for serial to connect
+  //while (!Serial) delay(1); // OPTIONAL: Wait for serial to connect
   Serial.println("*****Reflowduino demo*****");
 
   if (enableKeyboard) Keyboard.begin(); // Only if you want to type data into Excel
+  reflow = false;
+  buttonPressed = false;
+}
+
+void buttonPress() {
+  buttonPressed = true;
 }
 
+
 void loop() {
+
+  if (buttonPressed && (millis() - lastButtonPress) > BUTTON_DEBOUNCE_MSEC) {
+    lastButtonPress = millis();
+    justStarted = true;
```

Comment:
Great start on the buttons but I was thinking to implement this along with a display shield that has buttons on it. I don't see many users wiring buttons themselves without some sort of display along with it because they wouldn't be able to see the progress without the display anyway. Maybe we could create a definition at the top of the code which would allow the user to select if they have a shield or not, or maybe this would be better off as a completely separate set of code.